#!/usr/env/python
PACKAGE = "my_dyn_rec"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#List of parameters 
gen.add("position_x", double_t, 0, "center position x",  0.0, -200.0, 200.0) # default min max
gen.add("position_y", double_t, 0, "center position y",  0.0, -200.0, 200.0)
gen.add("position_z", double_t, 0, "center position y",  0.0, -200.0, 200.0)
gen.add("length_x", double_t, 0, "In meters",  0.0, 1.0, 20.0)
gen.add("length_y", double_t, 0, "In meters",  0.0, 1.0, 20.0)
gen.add("length_z", double_t, 0, "In meters",  0.0, 1.0, 20.0)
gen.add("topic_name_in" , str_t, 0,  "Input topic name. Restart needed if it changes.",  "/cloud")
gen.add("topic_name_out", str_t, 0, "Output topic name. Restart needed if it changes.",  "/filtered")
gen.add("debug", bool_t, 0, "publish debug topics too", True)

exit(gen.generate(PACKAGE, "my_dyn_rec", "MyParams"))